peers/app_request_network.go

```diff
@@ -94,38 +94,47 @@ func NewNetwork(
 		return nil, err
 	}
 
-	validatorClient := validators.NewCanonicalValidatorClient(logger, cfg.PChainAPI)
+	validatorClient := validators.NewCanonicalValidatorClient(logger, cfg.GetPChainAPI())
 
 	arNetwork := &AppRequestNetwork{
-		Network:         testNetwork,
-		Handler:         handler,
+		network:         testNetwork,
+		handler:         handler,
 		infoAPI:         infoAPI,
 		logger:          logger,
 		lock:            new(sync.Mutex),
 		validatorClient: validatorClient,
 	}
+	go logger.RecoverAndPanic(func() {
+		testNetwork.Dispatch()
```

Comment:
do we need to watch the error returned from Dispatch?

Reply:
I think that `RecoverAndPanic` should cover our needs here but could be wrong